community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyPayloadCursor.java

```diff
@@ -353,12 +352,4 @@ private static Object readArrayFromBuffer( ByteBuffer buffer )
             buffer.order( ByteOrder.LITTLE_ENDIAN );
         }
     }
-
-    private void assertOfType( PropertyType expected )
-    {
-        if ( type() != expected )
```

Comment:
Maybe replace this with assertion and keep all `#assertOfType()` as is to avoid message duplication?